### DISCUSSION TITLE
Force react-query cache to refetch in a test environment

### DISCUSSION THREAD

**Discussion by VanTanev** [TanStack/query#2564] [Q&A] [answered]
There are several cases in a normal environment where react-query will re-fetch data. How can I force that in a test environment?

More specifically, is there a way to force a refresh for a QueryObserver that is not used as a part of useQuery? I'm using a query observer as a sort of event emitter for a state machine, and want to force an update event.

**Comment by VanTanev** [comment, marked as answer]
This seems to work. Not sure if it's the optimal approach?
```ts
queryClient.resetQueries()
```

**Comment by TkDodo** [comment]
queryClient.refetchQueries() I’d say

**Reply by VanTanev** [reply]
In my case I needed to wrap that call in an act:
```ts
await act(() => queryClient.refetchQueries())

expect(screen.getByText(/new data/i)).toBeInTheDocument()
```

vs
```ts
queryClient.resetQueries()

await waitFor(() => expect(screen.getByText(/new data/i)).toBeInTheDocument())
```

Both are valid, depends on which you prefer.
